nautobot/extras/forms/mixins.py

```diff
@@ -137,6 +141,38 @@ def __init__(self, *args, **kwargs):
             self.custom_fields.append(field_name)
 
 
+class NoteFormBase(forms.Form):
+    """Base fore the NoteModelFormMixin and NoteModelBulkEditFormMixin."""
+
+    object_note = CommentField(label="Note")
+
+    def save_note(self, *, instance, user):
+        if "object_note" in self.cleaned_data and self.cleaned_data.get("object_note"):
+            value = self.cleaned_data.get("object_note")
+            note = Note.objects.create(
+                note=value,
+                assigned_object_type=self.obj_type,
+                assigned_object_id=instance.pk,
+                user=user,
+            )
+            logger.debug("Created %s", note)
```

Comment:
Perhaps avoid inadvertently creating whitespace-only Notes as well?

```suggestion
        value = self.cleaned_data.get("object_note", "").strip()
        if value:
            note = Note.objects.create(
                note=value,
                assigned_object_type=self.obj_type,
                assigned_object_id=instance.pk,
                user=user,
            )
            logger.debug("Created %s", note)
```